src/builtins.c

```diff
@@ -324,11 +332,10 @@ JL_DLLEXPORT uintptr_t jl_object_id_(jl_value_t *tv, jl_value_t *v) JL_NOTSAFEPO
     jl_datatype_t *dt = (jl_datatype_t*)tv;
     if (dt == jl_datatype_type) {
         jl_datatype_t *dtv = (jl_datatype_t*)v;
-        // `name->wrapper` is cacheable even though it contains TypeVars
-        // that don't have stable IDs.
-        //if (jl_egal(dtv->name->wrapper, v))
-        //    return bitmix(~dtv->name->hash, 0xaa5566aa);
-        return bitmix(~dtv->name->hash, hash_svec(dtv->parameters));
+        uintptr_t h = ~dtv->name->hash;
+        if (dtv->isconcretetype)
+            return bitmix(h, dtv->hash);
```

Comment:
Why isn't `dtv->hash` sufficient here?

Reply:
dtv->hash was only the contents of the parameters, so we need to add in the type itself now, but I could change that

Reply:
Yes, it seems to make sense to me to have the name's hash already mixed in. Any downside?